pkg/apis/aws/types_worker.go

```diff
@@ -117,3 +119,24 @@ const (
 	// VolumeTypeGP3 is a constant for the gp3 volume type.
 	VolumeTypeGP3 VolumeType = "gp3"
 )
+
+// HTTPTokensValue is a constant for HTTPTokens values.
+type HTTPTokensValue string
+
+const (
+	// HTTPTokensRequired is a constant for requiring the use of tokens to access IMDS. Effectively disables access via
+	// the IMDSv1 endpoints.
+	HTTPTokensRequired HTTPTokensValue = "required"
+	// HTTPTokensOptional that makes the use of tokens for IMDS optional. Effectively allows access via both IMDSv1 and
+	// IMDSv2 endpoints.
+	HTTPTokensOptional HTTPTokensValue = "optional"
+)
+
+// InstanceMetadataOptions contains configuration for controlling access to the metadata API.
+type InstanceMetadataOptions struct {
+	// HTTPTokens enforces the use of metadata v2 API.
+	HTTPTokens *HTTPTokensValue
+	// HTTPPutResponseHopLimit is the response hop limit for instance metadata requests.
+	// Valid values are between 1 and 64.
+	HTTPPutResponseHopLimit *int64
```

Comment:
If you have a separate field for disabling IMDS as commented above then you could not make it a pointer and default it to 1. Even aws does the same (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-metadataoptions.html)

Reply:
The idea behind exposing pointers is that when they are nil you allow AWS to default them. Case in point, if AWS would change their defaults to `2` then we wouldn't need to follow up with a mirroring change on our side, we simply don't assume what the default value is and don't pass this option if we don't have to. 

Our validation also would prevent you from using any value that doesn't fall in the `1-64` range anyway.

Reply:
> Case in point, if AWS would change their defaults to 2 then we wouldn't need to follow up with a mirroring change on our side, we simply don't assume what the default value is and don't pass this option if we don't have to.

Ok would it make sense to add this as part of the documentation that if you choose not to specify any value then provider defaults will be used (to make it clear that gardener does not default them to anything). This means that the consumer will have to be aware of the provider defaults which should ideally not be a problem.